### PR TITLE
add validation build and explicit deps

### PR DESCRIPTION
**Pull request by AndreyNikiforov**
use explicit dependencies for all libs and pip -- that will allow deterministic testing, including dependabot upgrade suggestions.

adding validation build as part of the quality checks, so we are sure that our release pipeline is executed on each push (except publishing). should help polishing release actions #224

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,14 +3,16 @@
 FROM python:3.9 as base
 
 WORKDIR /app
+# explicit requirements because runtime does not need ALL dependencies
+COPY requirements-pip.txt .
 COPY requirements.txt .
-RUN pip install -r requirements.txt
+RUN pip install -r requirements-pip.txt
```
Comment (menkej) [inline Dockerfile:9]: Is this a new file? Missing?

**Review by menkej — CHANGES_REQUESTED** [review]
Seems like a new file is introduced that is missing in the pull

File: Dockerfile (hunk 1 above)
Comment (AndreyNikiforov) [reply Dockerfile:9]: hmm. shows in the list of files in github UI for me

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,4 @@
 pyicloud_ipd==0.10.1
-#docopt==0.6.2
```
Comment (menkej) [inline requirements.txt:2]: I think that was a relic from the past anyways.

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
+setuptools==50.3.2
```
Comment (menkej) [inline requirements-dev.txt:1]: I still think we should only pin on minor releases... ;-)

File: .github/workflows/quality-checks.yml [hunk 4]
```diff
@@ -35,3 +35,45 @@ jobs:
...
+      uses: docker/setup-qemu-action@v1
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Build
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: ./Dockerfile
```
Comment (menkej) [inline .github/workflows/quality-checks.yml:77]: We are building releases with Dockerfile.release. Testing now with Dockerfile. Should this run `--target=test`?

What should happen here?

Comment (AndreyNikiforov) [reply .github/workflows/quality-checks.yml:81]: Good point. Adding release docker validation

File: requirements-dev.txt (hunk 3 above)
Comment (AndreyNikiforov) [reply requirements-dev.txt:1]: With dependabot the cost of upgrade is close to zero and pinning to patch version guarantees ~100% reproduction of the process.

**Comment by menkej** [comment]
while you're on it: what do you think about shortening the name of the jobs? `quality_checks_ubuntu_python`-> we know its ubuntu and python, so `quality_checks` is sufficient imho. shortens the tabs on the left in "Actions". What do you think?

**Comment by AndreyNikiforov** [comment]
> while you're on it: what do you think about shortening the name of the jobs? `quality_checks_ubuntu_python`-> we know its ubuntu and python, so `quality_checks` is sufficient imho. shortens the tabs on the left in "Actions". What do you think?

done

**Review by menkej — APPROVED** [review]

**Pull request merged**